src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java

```diff
@@ -535,7 +536,15 @@ void updateHostData() {
 
             Map<String, String> testMap = getForemanAPI().getCompatibleHosts();
             if (testMap != null) {
-                hostsMap.set(testMap);
+                // Randomize nodes ordering
```

Comment:
I am wondering if it would be more cost-effective to get a random key when we attempt to get the data as opposed to ALWAYS performing a shuffle on update()